### PR TITLE
Updating What's new pages

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/eyvLesm8/264-5-1-what-s-new-upgrade and various other doc cards on the 5.1 to do list.

Updating what's new features/breaking changes etc.

### CODE REVIEW

**Comment by hflynn** [comment]
Dev web stuff still to come

File: omero/developers/whatsnew.txt [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 Breaking changes
 ----------------
 
+OMERO.web
+^^^^^^^^^
+
+There have been many updates to this client for 5.1. See the
+:ref:`web-breaking-changes` section on the :doc:`/sysadmins/whatsnew` for
+sysadmins page for further details. Additional developer-specific changes 
+
```
Comment (mtbc) [inline omero/developers/whatsnew.txt]: still coming I guess?

File: omero/developers/whatsnew.txt [hunk 2]
```diff
@@ -1,16 +1,30 @@
 What's new for OMERO 5.1
 ========================
 
-OMERO 5.1 introduces units and some other stuff that others **will**
-write about.
-
+- :doc:`/sysadmins/version-requirements` provides extensive details about what
```
Comment (rleigh-dundee) [inline omero/developers/whatsnew.txt]: which rather than what?

File: omero/developers/whatsnew.txt [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+  support for Java 1.6 and Python 2.6.
 - :doc:`Server/ObjectGraphs` explains how the new graph operations of
   :javadoc:`omero::cmd <slice2html/omero/cmd.html>` are implemented.
-
+- Deprecated methods in IContainer and RenderingEngine have been removed, as
+  have the deprecated services IDelete and Gateway.
+- Pojos methods have been deprecated and replaced by corresponding units
+  methods.
+- OMERO 5.1 requires a Unicode-encoded (UTF8) database.
+- Transient-ness is now part of the public API.
```
Comment (rleigh-dundee) [inline omero/developers/whatsnew.txt]: Transiency?

Comment (mtbc) [inline omero/developers/whatsnew.txt]: even just transcience?

File: omero/sysadmins/whatsnew.txt [hunk 4]
```diff
@@ -1,14 +1,11 @@
...
 
-From a sysadmin's point of view, here is some introductory / overview
-text.
-
-- :doc:`server-upgrade` has **not yet** been updated to explain the
-  server upgrade process from OMERO 5.0 to OMERO 5.1.
-
-- :doc:`system-requirements` has been updated to reflect **various
-  things yet to be finalized**.
+- :doc:`version-requirements` provides extensive details about what operating
```
Comment (rleigh-dundee) [inline omero/sysadmins/whatsnew.txt]: which?

File: omero/developers/whatsnew.txt (hunk 3 above)
Comment (rleigh-dundee) [inline omero/developers/whatsnew.txt]: Trancience would be better, agreed.

**Comment by hflynn** [comment]
Should be all fixed.
@jburel @joshmoore you two should probably sign off on this

File: omero/developers/whatsnew.txt [hunk 5]
```diff
@@ -1,16 +1,33 @@
...
+  deprecating support for Java 1.6 and Python 2.6.
 - :doc:`Server/ObjectGraphs` explains how the new graph operations of
   :javadoc:`omero::cmd <slice2html/omero/cmd.html>` are implemented.
-
+- Deprecated methods in IContainer and RenderingEngine have been removed, as
+  have the deprecated services IDelete and Gateway.
+- Pojos methods have been deprecated and replaced by corresponding units
+  methods.
+- OMERO 5.1 requires a Unicode-encoded (UTF8) database.
+- Transience is now part of the public API.
```
Comment (joshmoore) [inline omero/developers/whatsnew.txt]: Aroo?

Comment (hflynn) [inline omero/developers/whatsnew.txt]: Not the most helpful comment ever! Do you want to explain the issue??

File: omero/sysadmins/whatsnew.txt [hunk 6]
```diff
@@ -1,14 +1,11 @@
...
-- :doc:`server-upgrade` has **not yet** been updated to explain the
-  server upgrade process from OMERO 5.0 to OMERO 5.1.
-
-- :doc:`system-requirements` has been updated to reflect **various
-  things yet to be finalized**.
+- :doc:`version-requirements` provides extensive details about which operating
+  systems and dependency versions we will be supporting for the life of 5.1
+  and the likely changes to these for the next major release (currently slated
+  to be 5.2). :doc:`system-requirements` has been updated to reflect these
+  updates.
```
Comment (manics) [inline omero/sysadmins/whatsnew.txt]: There's some discrepancies between system-requirements and version-requirements, e.g. Ubuntu 12.04 minimum, Ubuntu 14.04 recommended, Java 1.7 recommended, Python 2.7 recommended.

Comment (rleigh-dundee) [inline omero/sysadmins/whatsnew.txt]: Some discrepancies are inevitable given that an OS release aggregates many different packages with particular versions, but I'm not sure what the problem is here.  Both of these Ubuntu LTS releases provide both java 7 and python 2.7.  But we recommend the newer release over the older release.

Comment (manics) [inline omero/sysadmins/whatsnew.txt]: I meant system-requirements still says 10.04 is the minimum (should be 12.04) and recommends Java 1.6 / Python 2.6 (which are now the minimum).

Comment (hflynn) [inline omero/sysadmins/whatsnew.txt]: Thanks for spotting that Simon, I'll update it on #1172 since the other version updates are on that PR

Comment (rleigh-dundee) [inline omero/sysadmins/whatsnew.txt]: The system requirements are not solely based upon the version requirements though.  There's the question of how much we can realistically support and for how long.  In this case we've chosen to support the last two LTS releases.  While 10.04 will still work, it's in the "use at own risk" category: we have done *zero* building, testing and QA on this platform for 5.1. And support for the 5.1 deployment timeframe is also considered; for 10.04 LTS, support ends in April next month, making it unrealistic for a 5.1 deployment.

**Comment by hflynn** [comment]
Any further comments that actually relate to this PR?

File: omero/developers/whatsnew.txt (hunk 5 above)
Comment (mtbc) [inline omero/developers/whatsnew.txt]: I don't actually know what this new transience is but with luck those who should know do.

**Comment by rleigh-dundee** [comment]
@hflynn If we're dropping software version stuff from the system-requirements page, then maybe remove the system-requirements reference here?

**Comment by hflynn** [comment]
Oh, good point, will do.

File: omero/developers/whatsnew.txt (hunk 5 above)
Comment (joshmoore) [inline omero/developers/whatsnew.txt]: If this is related to the requirement for `Memoization`, then it'll need more explanation, perhaps a link? cc: @melissalinkert

Comment (melissalinkert) [inline omero/developers/whatsnew.txt]: I assume this is https://github.com/openmicroscopy/bioformats/pull/1362, from https://trello.com/c/YcdKrVc9/234-document-breaking-changes-for-5-1.

The only impact to external developers would be that someone writing a new reader needs to make sure that any global RandomAccessInputStream variables are marked as transient and initialized by the reopenFile() method in IFormatReader (```:javadoc: `reopenFile <loci/formats/IFormatReader.html#reopenFile>```` or similar might be the link to add).

Comment (hflynn) [inline omero/developers/whatsnew.txt]: Ok, so this doesn't really belong in the OMERO docs at all then? I don't think we really need to be linking to BF javadocs in the OMERO updates page.

**Comment by rleigh-dundee** [comment]
Thanks, looks good to merge.

**Comment by hflynn** [comment]
I'm just going to remove the transience point.

**Comment by hflynn** [comment]
--no-rebase

**Pull request merged**